src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -145,6 +157,19 @@ def updatePileup(self, doc, rseList=None, validate=True):
         if pname == '':
             pname = doc.get('pileupName', '')
             spec = {'pileupName': pname}
+
+        # add transition record if doc contains new fraction
+        if 'containerFraction' in doc:
+            transitionRecord = {
+                    'containerFraction': doc['containerFraction'],
+                    'customDID': customDID(pname),
+                    'updateTime': int(time.time()),
+                    'DN': userDN}
+            transition = []
+            if 'transition' in doc:
+                transition = doc['transition']
+            transition.append(transitionRecord)
```

Comment:
When is `transition` variable used? I would suggest to replace these 4 lines of code by something like:
```
doc.setdefault('transition', [])
doc['transition'].append(transitionRecord)
```

Reply:
done